src/Helpers/ZStdHelper.cs

```diff
@@ -53,4 +66,31 @@ public static void DecompressFolder(string path, string output, bool recursive,
             updateCount?.Invoke(i + 1);
         }
     }
+
+    public static void CompressFolder(string path, string output, bool recursive, Action<int>? setCount = null, Action<int>? updateCount = null)
+    {
+        string[] allFiles = Directory.GetFiles(path, "*.*", recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly);
+        List<string> files = new(allFiles.Length);
```

Comment:
This is unnecessary, the extension can be checked in the for loop avoiding iterating the files twice.

```cs
for (int i = 0; i < files.Count; i++)
{
    string file = files[i];
    if (Path.GetExtension(file) == ".zs") {
        continue;
    }
    
    Span<byte> data = Compress(file);

    string outputFile = Path.Combine(output, Path.GetRelativePath(path, $"{file}.zs"));
    Directory.CreateDirectory(Path.GetDirectoryName(outputFile)!);
    using FileStream fs = File.Create(outputFile);
    fs.Write(data);

    updateCount?.Invoke(i + 1);
}
```